bravenewcoin-vwap/adapter.js

```diff
@@ -0,0 +1,96 @@
+const { Requester, Validator } = require('@chainlink/external-adapter')
+
+const customParams = {
+  symbol: ['base', 'from', 'coin', 'symbol', 'assetId', 'indexId', 'asset'],
+  indexType: false,
+  timestamp: false
+}
+
+const host = 'bravenewcoin.p.rapidapi.com'
+
+const authenticate = () => {
+  return new Promise((resolve, reject) => {
+    Requester.request({
+      method: 'POST',
+      url: `https://${host}/oauth/token`,
+      headers: {
+        'content-type': 'application/json',
+        'x-rapidapi-host': host,
+        'x-rapidapi-key': process.env.API_KEY,
+        accept: 'application/json',
+        useQueryString: true
+      },
+      data: {
+        audience: 'https://api.bravenewcoin.com',
+        client_id: process.env.CLIENT_ID,
+        grant_type: 'client_credentials'
+      }
+    }).then(response => resolve(response.data.access_token))
+      .catch(error => reject(error))
+  })
+}
+
+const getAssetId = (token, symbol) => {
+  return new Promise((resolve, reject) => {
+    Requester.request({
+      url: `https://${host}/asset`,
+      headers: {
+        'content-type': 'application/octet-stream',
+        'x-rapidapi-host': host,
+        'x-rapidapi-key': process.env.API_KEY,
+        useQueryString: true
+      },
+      params: {
+        status: 'ACTIVE',
+        symbol
+      }
+    }).then(response => resolve(response.data.content[0].id))
+      .catch(error => reject(error))
+  })
+}
+
+const createRequest = (input, callback) => {
```

Comment:
As our `@chainlink/ea-bootstrap` calls the `createRequest` function directly and expects a callback, to bridge the API, we could do something like the following.

Create an async `_createRequest` function that has all the logic, simplified using `async/await`:

```javascript
const _createRequest = async (input) => {
  ...
  return {
    status: response.status, 
    data: Requester.success(jobRunID, response)
  }
}
```

And then the `createRequest` would just bridge the async/callback API:

```javascript
const createRequest = (input, callback) => {
  _createRequest(input)
    .then(result => callback(result.status, result.data))
    .catch(error => callback(500, Requester.errored(jobRunID, error)))
}